NetworkOfPrivateClinics/NetworkOfPrivateClinics/BusinessLogic/ClinicRepository.cs

```diff
@@ -6,27 +6,36 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace NetworkOfPrivateClinics.BisinessLogic
+namespace NetworkOfPrivateClinics.BusinessLogic
 {
     public class ClinicRepository : IClinicRepository
     {
-        private readonly List<Clinic> _context;
+        private readonly List<Clinic> _clinics = new();
 
-        public ClinicRepository(List<Clinic> clinicContext)
+        public async Task AddClinicAsync(Clinic clinic)
         {
-            _context = clinicContext;
+            await Task.Run(() =>
+            {
+                _clinics.Add(clinic);
+            });
         }
 
-        public void DeleteClinic(int id)
+        public async Task DeleteClinicAsync(Clinic clinic)
         {
-            Clinic clinic = _context.First(x => x.ClinicID == id);
-            _context.Remove(clinic);
+            await Task.Run(() =>
+            {
+                _clinics.Remove(clinic);
+            });
         }
```

Comment:
No needed. Why would u make it async if u work with in memory storage. It's not I/O bound operation